src/ORM/ArrayList.php

```diff
@@ -126,9 +126,13 @@ public function exists()
      */
     public function getIterator(): Traversable
     {
-        foreach ($this->items as $i => $item) {
+        foreach ($this->items as $item) {
             if (is_array($item)) {
-                yield new ArrayData($item);
+                if (array_is_list($item)) {
+                    yield new ArrayList($item);
+                } else {
+                    yield new ArrayData($item);
+                }
```

Comment:
I don't think we should be changing the way `ArrayList` works - people use `ArrayList` for purposes other than just use in templates, and this could break some logic people have that assumes the current logic will remain as-is.